frame/bridge/ethereum/issuing/src/lib.rs

```diff
@@ -21,275 +21,347 @@
+		pub fn set_ethereum_backing_address(
+			origin: OriginFor<T>,
+			address: EthereumAddress,
+		) -> DispatchResultWithPostInfo {
+			ensure_root(origin)?;
+			let old_address = EthereumBackingAddress::<T>::get();
+			EthereumBackingAddress::<T>::put(address);
+			Self::deposit_event(Event::EthereumBackingAddressUpdated(old_address, address));
+			Ok(().into())
+		}
 	}
+}
 
-	fn abi_encode_token_redeem(result: EthLog) -> Result<Vec<u8>, DispatchError> {
-		log::debug!("abi_encode_token_redeem");
-		// parse the following ethereum backing lock event
-		// BackingLock(address indexed sender, address source, address target, uint256 amount, address receiver, uint256 fee)
-		// @sender & @source are not used here
-		// @target(params[2]): the mapped token address
-		// @amount(params[3]): the token amount [wei]
-		// @receiver(params[4]): the dvm receiver address
-		// @fee(params[5]): the fee for this cross transfer
-		let dtoken_address = result.params[2]
-			.value
-			.clone()
-			.into_address()
-			.ok_or(<Error<T>>::AddressCF)?;
-		let amount = result.params[3]
-			.value
-			.clone()
-			.into_uint()
-			.ok_or(<Error<T>>::UintCF)?;
-		let recipient = result.params[4]
-			.value
-			.clone()
-			.into_address()
-			.ok_or(<Error<T>>::AddressCF)?;
-		let fee = result.params[5]
-			.value
-			.clone()
-			.into_uint()
-			.ok_or(<Error<T>>::UintCF)?;
-
-		let input = Abi::encode_cross_receive(dtoken_address, recipient, amount)
-			.map_err(|_| Error::<T>::InvalidMintEncoding)?;
-
-		log::debug!("transfer fee will be delived to fee pallet {}", fee);
-		Ok(input)
+impl<T: Config> Pallet<T> {
+	pub fn digest() -> PalletDigest {
+		let mut digest: PalletDigest = Default::default();
+		let pallet_digest = sha3::Keccak256::digest(T::PalletId::get().encode().as_slice());
+		digest.copy_from_slice(&pallet_digest[..4]);
+		digest
 	}
 
 	pub fn mapped_token_address(
 		backing: EthereumAddress,
 		source: EthereumAddress,
-	) -> Result<H160, DispatchError> {
-		let factory_address = MappingFactoryAddress::get();
-		let bytes = Abi::encode_mapping_token(backing, source)
+	) -> Result<EthereumAddress, DispatchError> {
+		let factory_address = MappingFactoryAddress::<T>::get();
+		let bytes = mtf::encode_mapping_token(backing, source)
 			.map_err(|_| Error::<T>::InvalidIssuingAccount)?;
 		let mapped_address = dvm_ethereum::Pallet::<T>::do_call(factory_address, bytes)
 			.map_err(|e| -> &'static str { e.into() })?;
 		if mapped_address.len() != 32 {
 			return Err(Error::<T>::InvalidAddressLen.into());
 		}
-		Ok(H160::from_slice(&mapped_address.as_slice()[12..]))
+		Ok(EthereumAddress::from_slice(
+			&mapped_address.as_slice()[12..],
+		))
 	}
 
 	pub fn finish_token_registered(
 		backing: EthereumAddress,
 		source: EthereumAddress,
 		target: EthereumAddress,
 	) -> DispatchResult {
-		let raw_event = RawEvent::TokenRegistered(0, backing, source, target);
+		let raw_event = Event::TokenRegistered(0, backing, source, target);
```

Comment:
`Event::TokenRegistered(0, backing, source, target);` Make this `0` to a const!!